src/lib/accessibility/Behaviors/Button/ButtonBehavior.ts

```diff
@@ -1,18 +1,20 @@
-import { Accessibility } from '../../interfaces'
+import { Accessibility, AccessibilityAttributes } from '../../interfaces'
 
 /**
  * @description
  *  Adds role='button' if element type is other than 'button'. This allows screen readers to handle the component as a button.
  *  Adds attribute 'aria-disabled=true' based on the property 'disabled'.
  */
 
-const ButtonBehavior: Accessibility = (props: any) => ({
-  attributes: {
-    root: {
-      role: props.as === 'button' ? undefined : 'button',
-      'aria-disabled': !!props['disabled'],
-    },
+export const ButtonAttributes = (props: any): AccessibilityAttributes => ({
```

Comment:
How does this help with reusability? If `ToggleButtonBehavior` is built on top of `ButtonBehavior`, it can call `ButtonBehavior` directly:
```jsx
  const parent = ButtonBehavior(props)
  return {
    attributes: {
      root: {
        ...parent.attributes.root,
        'aria-pressed': !!props['active'],
        'aria-disabled': !!props['disabled'],
      },
    },
  }
```

Extracting to `ButtonAttributes` can be understood as *only behaviors with extracted attributes can be 'extended'* which (1) is not true and (2) limits extending behaviors for library users.

Reply:
@miroslavstastny agree with your point that it is possible to that as in your example, although I wouldn't call it "parent". It is like a composition - you can build new behavior with different parts. + in your approach the whole behavior is initialized and we use from it just part, so what can have a bigger performance impact.
Let me know what you think.

Reply:
Understand, but do we need it now or is it just a premature optimisation? Especially considering that currently the only refactored behavior is `ButtonBehavior` which has just `attributes` and nothing else.
I think that your approach is valid when it is required but I would not define it as the only/primary reusability pattern.

Reply:
Don't take me wrong, I like the change to functions.
I would just prefer not to have `ButtonAttributes`